File: js_snowball/src/Among.js
/*!
 * Snowball JavaScript Library v0.4
 * http://snowball.tartarus.org/
 * http://nongreedy.ru/
 *
 * Copyright Nov 25 2012, Oleg Mazko
 * http://creativecommons.org/licenses/by/3.0/us/
 */

function Among(s, substring_i, result, method) {
	if ((!s && s != "") || (!substring_i && (substring_i != 0)) || !result)
		throw ("Bad Among initialisation: s:" + s + ", substring_i: "
				+ substring_i + ", result: " + result);
	this.s_size = s.length;
	this.s = this.toCharArray(s);
	this.substring_i = substring_i;
	this.result = result;
	this.method = method;
}
Among.prototype.toCharArray = function(s) {
	var sLength = s.length, charArr = new Array(sLength);
	for (var i = 0; i < sLength; i++)
		charArr[i] = s.charCodeAt(i);
	return charArr;
}

File: js_snowball/eclipse/src/org/tartarus/snowball/Among.java
package org.tartarus.snowball;

import java.lang.reflect.Method;

public class Among {

	private static char[] toCharArray(String s) {
		int sLength = s.length();
		char[] charArr = new char[sLength];
		for (int i = 0; i < sLength; i++)
			charArr[i] = s.charAt(i);
		return charArr;
	}

	// :es6:
	// remove this ctor
	public Among(String s, int substring_i, int result) {
		this.s = toCharArray(s);
		this.substring_i = substring_i;
		this.result = result;
		this.method = null;
	}
	// :end:

	public Among(String s, int substring_i, int result, String methodname, Class obj) {
		this.s = toCharArray(s);
		this.substring_i = substring_i;
		this.result = result;
		// :es6:
		// method = methodname ? obj[methodname] : null;
		// methodobject = obj;
		try {
			this.method = obj.getDeclaredMethod(methodname);
		} catch (NoSuchMethodException e) {
			throw new RuntimeException(e);
		}
		// :end:
	}

	public final char[] s; /* search string */
	public final int substring_i; /* index to longest matching substring */
	public final int result; /* result of the lookup */
	public final Method method; /* method to use if substring matches */
	// :es6:
	public Object methodobject; /* object to invoke method on */
	// :end:
};
